qcodes/dataset/measurements.py

```diff
@@ -18,6 +18,9 @@
 
 log = logging.getLogger(__name__)
 
+array_like_types = (tuple, list, np.ndarray)
```

Comment:
Maybe its better to do something like: 

```python
import collections

if isinstance(value, collections.Iterable): 
   ....
```

Reply:
Note that that will also include strings which we probably dont want to consider arrays of chars